File: Easy/1704. Determine if String Halves Are Alike/main.rb
# @param {String} s
# @return {Boolean}
def halves_are_alike(s)
    half = s.length / 2
    count = 0
    for i in 0..half-1
        count += 1 if 'ueoai'.include?(s[i].downcase)
        count -= 1 if 'ueoai'.include?(s[i + half].downcase)
    end
    return count == 0
end

File: Easy/1704. Determine if String Halves Are Alike/main.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        half = len(s) // 2
        count = 0
        for i in range(half):
            if s[i].lower() in 'ueoai': count += 1 # first half
            if s[i + half].lower() in 'ueoai': count -= 1 # second half
        return count == 0
